### ISSUE TITLE
face a problem of the loss of training is nan

### ISSUE THREAD

**Issue by yansuoyuli** [yyh-rain-song/ReMamber#3]
Thanks for the awesome work. But I face  a problem of the loss of training is nan during training many times . I want to ask how to solve this problem.

**Comment by yansuoyuli** [comment]
![截屏2024-10-20 12 52 44](https://github.com/user-attachments/assets/9c60283d-88e6-4947-805b-60cf585556ea)

**Comment by yansuoyuli** [comment]
And I also want to ask how the mix the three datasets during training: 'refcoco', 'refcoco+', 'refcocog'

**Comment by yyh-rain-song** [comment]
Loss NaN can often be related to the learning rate or batch size settings. In our experiments, we set batch_size=8 and trained on 8 GPUs (total batch_size=64), using the default learning rate provided in the code. We didn't encounter any NaN loss issues with this setup.

If you're experiencing NaN loss, it might indicate that you need to either reduce the learning rate or increase the batch size. Adjusting these settings may help resolve the issue.

**Comment by yyh-rain-song** [comment]
As for mixing the three datasets, the mixing process is simply a concatenation of the datasets, where the three datasets are combined in a 1:1:1 ratio and then randomly shuffled during training. For simplicity we removed the mixed dataset training logic from the current code, but you can easily implement this yourself.

Hope this clarifies things!
